secret_provider/src/k8s.rs

```diff
@@ -0,0 +1,82 @@
+//! This module reads from a secret provided the given key matches. Consider the example as below.
+//!
+//! apiVersion: v1
+//! kind: Secret
+//! metadata:
+//!   name: my-secret
+//!   namespace: default
+//! type: Opaque
+//! stringData:
+//!   encryption_parameters: |
+//!    {
+//!       "cipher": "AesCbc",
+//!       "key": {
+//!        "key_name": "example-key",
+//!        "key": [115, 111, 109, 101, 45, 107, 101, 121],
+//!         "key_length": 128,
+//!         "key2": null,
+//!         "key2_length": null
+//!       }
+//!     }
+//!
+use crate::types::EncryptionParameters;
+use crate::SecretProvider;
+
+use crate::error::Error;
+use async_trait::async_trait;
+use k8s_openapi::api::core::v1::Secret;
+
+const SECRET_DATA_KEY: &str = "encryption_parameters";
+
+/// A secret provider that retrieves a Kubernetes secret.
+pub struct K8sSecretProvider {
+    /// The Kubernetes client.
+    pub client: kube::Client,
+    /// The namespace in which to look for the secret.
+    pub namespace: String,
```

Comment:
You could take it from the client default namespace?

Reply:
That might not always work, if it's a non mayastor namespace

Reply:
It's supposed to be passed down by the service that is using this crate, as it would in the same ns.